lib/rules/no-plusplus.js

```diff
@@ -40,18 +40,36 @@ module.exports = {
             allowInForAfterthought = config.allowForLoopAfterthoughts === true;
         }
 
+        const sourceCode = context.getSourceCode();
+
         return {
 
             UpdateExpression(node) {
                 if (allowInForAfterthought && node.parent.type === "ForStatement") {
                     return;
                 }
+
+                let fixerFunc;
+
+                if (node.parent.type === "ExpressionStatement" || node.parent.type === "ForStatement" || node.prefix) {
```

Comment:
I think this check isn't strict enough; it allows things like this:

```js
for (var foo = bar++; bar; baz) {} // `foo` will start out with the wrong value
```

To fix this, I would recommend changing the check a bit so that `++` expressions are only allowed in the "update" of the `for` loop (the last clause):

```js
if (
    node.parent.type === "ExpressionStatement" ||
    node.prefix ||
    (node.parent.type === "ForStatement" && node === node.parent.update)
) {
  // do a fix
}
```